lib/src/puppetdb-cli.cc

```diff
@@ -116,6 +208,54 @@ size_t write_body(char *ptr, size_t size, size_t nmemb, void *userdata){
     return size * nmemb;
 }
 
+class JsonQueueWrapper {
+  public:
+    typedef char Ch;
+    JsonQueueWrapper(JsonQueue& stream) : stream_(stream) {}
+    Ch Peek() const {  // 1
+        int c = Front();
+        return c == char_traits<char>::eof() ? '\0' : (Ch)c;
+    }
+    Ch Take() {  // 2
+        int c = Get();
+        return c == char_traits<char>::eof() ? '\0' : (Ch)c;
+    }
+    size_t Tell() const { return (size_t)stream_.size(); }  // 3
+    Ch* PutBegin() { assert(false); return 0; }
```

Comment:
They are common in the rapidjson wrappers, we will never hit this code, if we do something has gone horribly wrong with rapidjson or we're making a pretty big error, http://en.cppreference.com/w/cpp/error/assert